nipoppy/workflows/pipeline_store/install.py

```diff
@@ -98,6 +105,58 @@ def _update_config_and_save(self, pipeline_config: BasePipelineConfig) -> Config
 
         return config
 
+    def _download_container(self, pipeline_config: BasePipelineConfig):
+        # pipeline is not containerized
+        if pipeline_config.CONTAINER_INFO.URI is None:
+            return
+
+        # apply substitutions
+        pipeline_config = BasePipelineConfig(
+            **apply_substitutions_to_json(
+                pipeline_config.model_dump(mode="json"), self.config.SUBSTITUTIONS
+            )
+        )
+        fpath_container = Path(
+            process_template_str(
+                str(pipeline_config.get_fpath_container()), objs=[self.layout]
+            )
+        )
+
+        # container file already exists
+        if fpath_container.exists():
+            return
+
+        # prompt user and confirm
+        if self.assume_yes or Confirm.ask(
+            (
+                "[yellow]This pipeline is containerized: do you want to download the "
+                f"container (to [magenta]{fpath_container}[/]) now?[/]"
+            ),
+            default=True,
+        ):
+            try:
+                # use stderr for status messages so that the Apptainer/Singularity
+                # output does not break the status display
+                # ("apptainer/singularity pull" seems to only print to stderr)
+                with Console().status(
+                    "Downloading the container, this can take a while..."
+                ) as status:
+                    self.run_command(
+                        [
+                            self.config.CONTAINER_CONFIG.COMMAND,
+                            "pull",
+                            fpath_container,
+                            pipeline_config.CONTAINER_INFO.URI,
+                        ]
+                    )
+                    status.update(f"Container downloaded to {fpath_container}")
+            except subprocess.CalledProcessError as exception:
+                self.logger.error(
+                    f"Failed to download container {pipeline_config.CONTAINER_INFO.URI}"
+                    f" to {fpath_container}: {exception}"
+                )
+                self.return_code = ReturnCode.UNKNOWN_FAILURE
```

Comment:
Do we still consider the pipeline installation as successful if the container cannot download?
If not, we might want to raise SystemExit instead.